plugins/languages/typst/typst-vim.nix

```diff
@@ -0,0 +1,42 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.typst-vim;
+  helpers = import ../../helpers.nix {inherit lib;};
+in {
+  options.plugins.typst-vim =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "typst.vim";
+
+      package = helpers.mkPackageOption "typst-vim" pkgs.vimPlugins.typst-vim;
+
+      keymaps = {
+        silent = mkOption {
+          type = types.bool;
+          description = "Whether typst-vim keymaps should be silent.";
+          default = false;
+        };
+
+        watch =
+          helpers.mkNullOrOption types.str
+          "Keymap to preview the document and recompile on change.";
+      };
+    };
+
+  config = mkIf cfg.enable {
+    extraPlugins = [cfg.package];
+
+    # Add the typst compiler to nixvim packages
+    extraPackages = with pkgs; [typst];
+
+    maps.normal = with cfg.keymaps;
+      helpers.mkModeMaps {inherit silent;} {
+        ${watch} = ":TypstWatch<CR>";
```

Comment:
Does this work when watch `watch` is null ? I'm not sure

Reply:
Yes it works. Cf the `empty` test case.
I also tested locally without any mapping.
However, maybe it is overkill to wrap keybindings for this plugin. It does not really add any value...

Reply:
Ah sorry I though the keymaps were optional. It's not really a problem to provide keymaps, it helps discovery of options.